### PR TITLE
Add test that uses localization

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/grpc/grpc-dotnet/issues/2024

### CODE REVIEW

File: test/FunctionalTests/Client/UnaryTests.cs [hunk 1]
```diff
@@ -68,6 +68,35 @@ Task<HelloReply> UnaryThrowError(HelloRequest request, ServerCallContext context
...
+        }
+
+        // Arrange
+        var method = Fixture.DynamicGrpc.AddUnaryMethod<HelloRequest, HelloReply>(UnaryThrowError);
+        var channel = CreateChannel();
+        var client = TestClientFactory.Create(channel, method);
+        var metadata = new Metadata();
+        if (clientAcceptLanguage != null)
+        {
+            metadata.Add("accept-language", clientAcceptLanguage);
```
Comment (mgravell) [inline test/FunctionalTests/Client/UnaryTests.cs:89]: API thought unrelated to this PR, but more to the feature; should this be / is this already automated at the client? `metadata.AddRequestLocalization()`? an interceptor? (not sure either works perfectly; just asking the question)

File: test/FunctionalTests/Client/UnaryTests.cs [hunk 2]
```diff
@@ -68,6 +68,35 @@ Task<HelloReply> UnaryThrowError(HelloRequest request, ServerCallContext context
         StringAssert.StartsWith("Failed to deserialize response message.", call.GetStatus().Detail);
     }
 
+    [TestCase("fr", "fr")]
```
Comment (mgravell) [inline test/FunctionalTests/Client/UnaryTests.cs:71]: IMO we should have a test of what happens if the culture requested is not supported by the server

**Review by mgravell — APPROVED** [review]

File: test/FunctionalTests/Client/UnaryTests.cs (hunk 2 above)
Comment (JamesNK) [reply test/FunctionalTests/Client/UnaryTests.cs:71]: That's testing the localization feature. These tests are to double-check that gRPC doesn't lose the culture that localization sets.

File: test/FunctionalTests/Client/UnaryTests.cs (hunk 1 above)
Comment (JamesNK) [reply test/FunctionalTests/Client/UnaryTests.cs:89]: No other gRPC clients set this automatically.

**Pull request merged**